test/functional/example_test.py

```diff
@@ -203,7 +202,7 @@ def run_test(self):
 
         # wait_until() will loop until a predicate condition is met. Use it to test properties of the
         # P2PInterface objects.
-        wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2p.block_receive_map.keys())), timeout=5, lock=mininode_lock)
+        self.wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2p.block_receive_map.keys())), timeout=5, lock=mininode_lock)
```

Comment:
```suggestion
        self.nodes[2].p2p.wait_until(lambda: sorted(blocks) == sorted(list(self.nodes[2].p2p.block_receive_map.keys())), timeout=5)
```

Shouldn't this be

Reply:
Is there a big difference between two? I was inclined to use the one from framework wherever possible.

Reply:
It seems a bit convolved to import the lock, which needs to be taken, then call a method, which passes the imported lock to be taken.

Simply calling a member function that takes the lock when needed without importing or passing it seems less error prone, as there is no way it can be missed. Also, it makes the calling code shorter and more readable

Reply:
Updated, thanks for clear explanation.